purchase_open_qty/models/purchase_order.py

```diff
@@ -0,0 +1,72 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Eficent Business and IT Consulting Services S.L.
+#   (http://www.eficent.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from openerp import api, fields, models
+
+
+class PurchaseOrderLine(models.Model):
+    _inherit = 'purchase.order.line'
+
+    @api.depends('invoice_lines.invoice_id.state', 'product_qty')
+    def _compute_qty_invoiced(self):
+        super(PurchaseOrderLine, self)._compute_qty_invoiced()
+        for line in self:
+            line.qty_to_invoice = line.product_qty - line.qty_invoiced
+
+    @api.depends('order_id.state', 'move_ids.state', 'product_qty')
+    def _compute_qty_received(self):
+        super(PurchaseOrderLine, self)._compute_qty_received()
+        for line in self:
+            line.qty_to_receive = line.product_qty - line.qty_received
+
+    qty_to_invoice = fields.Float(compute='_compute_qty_invoiced',
+                                  string="Qty to invoice", store=True)
+    qty_to_receive = fields.Float(compute='_compute_qty_to_receive',
+                                  string="Qty to receive", store=True)
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    def _compute_qty_to_invoice(self):
+        for po in self:
+            po.qty_to_invoice = 0.0
+            for po_line in po.order_line:
+                po.qty_to_invoice += po_line.qty_to_invoice
+
+    def _compute_qty_to_receive(self):
+        for po in self:
+            po.qty_to_receive = 0.0
+            for po_line in po.order_line:
+                po.qty_to_receive += po_line.qty_to_receive
+
+    @api.model
+    def _search_qty_to_invoice(self, operator, value):
+        po_line_obj = self.env['purchase.order.line']
+        res = []
+        po_lines = po_line_obj.search(
+            [('qty_to_invoice', operator, value)])
+        order_ids = po_lines.mapped('order_id')
+        res.append(('id', 'in', order_ids.ids))
+        return res
+
+    @api.model
+    def _search_qty_to_receive(self, operator, value):
+        po_line_obj = self.env['purchase.order.line']
+        res = []
+        po_lines = po_line_obj.search(
+            [('qty_to_receive', operator, value)])
+        order_ids = po_lines.mapped('order_id')
+        res.append(('id', 'in', order_ids.ids))
+        return res
+
+    qty_to_invoice = fields.Float(compute='_compute_qty_to_invoice',
+                                  search='_search_qty_to_invoice',
+                                  string="Has qty pending to invoice",
```

Comment:
Since this is not a boolean anymore, 'Qty to invoice' is going to be a better name.